python/cuml/manifold/umap.pyx

```diff
@@ -41,26 +38,17 @@ cp_coo_matrix = gpu_only_import_from('cupyx.scipy.sparse', 'coo_matrix')
 cp_csc_matrix = gpu_only_import_from('cupyx.scipy.sparse', 'csc_matrix')
 
 import cuml.internals
-from cuml.common import using_output_type
 from cuml.internals.base import UniversalBase
 from pylibraft.common.handle cimport handle_t
 from cuml.common.doc_utils import generate_docstring
 from cuml.internals import logger
 from cuml.internals.input_utils import input_to_cuml_array
-from cuml.internals.memory_utils import using_output_type
-from cuml.internals.import_utils import has_scipy
 from cuml.internals.array import CumlArray
 from cuml.internals.array_sparse import SparseCumlArray
 from cuml.internals.mixins import CMajorInputTagMixin
 from cuml.common.sparse_utils import is_sparse
 from cuml.metrics.distance_type cimport DistanceType
 
-from cuml.manifold.simpl_set import fuzzy_simplicial_set, \
-    simplicial_set_embedding
```

Comment:
Deleting the import of `simplicial_set_embedding` broke `test_simpl_set.py`. Is this meant to be a public API exposed from `cuml.manifold.umap`? Do we need to worry about other unused imports that are meant to be exposed as public APIs? If so, the proper mechanism is to define `__all__ = ["simplicial_set_embedding", ...]`.

cc: @wphicks @dantegd @csadorf

Reply:
Thanks @csadorf!

Reply:
Yes, please go ahead with that.

Reply:
I filed an issue about this: https://github.com/MarcoGorelli/cython-lint/issues/79

Reply:
`cython-lint` also doesn't detect usage in `__all__`, so that isn't a viable solution here. I filed a second issue: https://github.com/MarcoGorelli/cython-lint/issues/80

Reply:
For now, I added these imports back with `# no-cython-lint` in d773305. That should resolve the problem. I reviewed the changes in this PR for other possible public API breaks but I didn't see any candidates except for https://github.com/rapidsai/cuml/pull/5439#discussion_r1248133315.